src/components/Timer.jsx

```diff
@@ -0,0 +1,74 @@
+import { useTimer } from 'react-timer-hook'
+import { useRecoilState } from 'recoil'
+import { combineState, currentTimerState, timerState } from '../atom'
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+
+export function Timer({ expiryTimestamp, autoStart = false, id, name, onClick }) {
+	const { seconds, minutes, hours, days, isRunning, start, pause, resume, restart } = useTimer({
+		expiryTimestamp,
+		onExpire: () => console.warn('onExpire called'),
+		autoStart,
+	})
+	const [timers, setTimers] = useRecoilState(timerState)
+	const [currentTimer, setCurrentTimer] = useRecoilState(currentTimerState)
+
+	useEffect(() => {
+		restart(expiryTimestamp, false)
+	}, [expiryTimestamp])
+
+	useEffect(() => {
+		setTimers({ ...timers, [id]: { name, time: hours * 60 * 60 + minutes * 60 + seconds } })
+	}, [hours, minutes, seconds])
+
+	useEffect(() => {
+		if (currentTimer.id !== id) pause()
+	}, [currentTimer])
+
+	return (
+		<TimerWrapper
+			id={id}
+			onClick={() => {
+				if (onClick === undefined) {
+					if (!isRunning) {
+						resume()
+						setCurrentTimer({ id, name })
+					} else {
+						pause()
+						setCurrentTimer({ id: null, name: null })
+					}
+				} else {
+					onClick()
+				}
+			}}
```

Comment:
if 문 중첩은 코드를 파악하려면 생각을 많이해야하는 것 같아요. 
`early return`을 사용해보는건 어때요?
요런식으로도 할 수 있을 것 같네요! 
onClick은 사실 왜 분기문이 필요한지 잘 모르겠는데 용도가 무엇인가요?? 👀

```suggestion
			onClick={() => {
				if (onClick !== undefined) {
					onClick()
					return
				}
				if (isRunning){
					pause()
					setCurrentTimer({ id: null, name: null })
					return;
				}
				resume()
				setCurrentTimer({ id, name })
			}}

```